### PR TITLE
Rich text node

### PR DESCRIPTION
**Pull request by fakob**
Added a rich text node. It is based on [slatejs](https://docs.slatejs.org/general/resources#products), which is a basic framework for text editors used by for example Discord and Grafana. It does not come with much, so you have to unfortunately build a lot yourself. On the other hand, it is therefore very flexible. In our implementation you can
* set background color
* change formatting of the text using the hover toolbar (shows up when something is selected)
  * keyboard shortcuts similar to google docs
    * links - alt+click to open in tab
* copy paste html into the editor (basic support)
* render input values. by typing @ it lets you select from a list of available input sockets. An input token is inserted into the text. After clicking outside, the tokens are replaced with whatever is plugged in
  * one input is there by default
  * you can either right click the node and add more or type @ and add the input directly from the editor

Screenshot 2022-09-18 at 19 08 50

Screenshot 2022-09-18 at 19 12 03

### CODE REVIEW

**Comment by fakob** [comment]
@TobiasElinder I found a couple of smaller bugs which I want to fix first. I will ping you again for a review.

**Comment by fakob** [comment]
@TobiasElinder I have submitted some fixes, could you please review?

File: package.json [hunk 1]
```diff
@@ -54,6 +55,10 @@
     "react-error-boundary": "^3.1.4",
     "react-lazylog": "^4.5.3",
     "react-monaco-editor": "^0.50.1",
+    "slate": "^0.82.1",
+    "slate-history": "^0.66.0",
+    "slate-hyperscript": "^0.77.0",
+    "slate-react": "^0.82.1",
```
Comment (magnificus) [inline package.json:61]: these are a lot of new dependencies :/, well I guess if we need to we need to

**Review by magnificus — APPROVED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -568,52 +567,53 @@ Viewport position (scale): ${viewportScreenX}, ${Math.round(
...
           e.preventDefault();
           currentGraph.current.selection.selectAllNodes();
         }
-        if ((isMac ? e.metaKey : e.ctrlKey) && e.key === 'f') {
+        if (modKey && e.key === 'f') {
           e.preventDefault();
           openNodeSearch(mousePosition);
         }
-        if ((isMac ? e.metaKey : e.ctrlKey) && e.key === 'd') {
+        if (modKey && e.key === 'd') {
```
[the remark this answers is not in the document]
Comment (fakob) [reply src/App.tsx]: Good idea

**Pull request merged**